File: Serene/Serene.Script/Imports/FormContexts/Membership.SignupForm.cs

namespace Serene.Membership
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    public partial class SignUpForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Membership.SignUp";
    
        public SignUpForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor DisplayName { get { return ById<StringEditor>("DisplayName"); } }
        public EmailEditor Email { get { return ById<EmailEditor>("Email"); } }
        public EmailEditor ConfirmEmail { get { return ById<EmailEditor>("ConfirmEmail"); } }
        public PasswordEditor Password { get { return ById<PasswordEditor>("Password"); } }
        public PasswordEditor ConfirmPassword { get { return ById<PasswordEditor>("ConfirmPassword"); } }
    }
}

File: Serene/Serene.Core/Imports/ServerTypings/Membership.SignUpForm.ts
namespace Serene.Membership {
    export interface SignUpForm {
        DisplayName: Serenity.StringEditor;
        Email: Serenity.EmailEditor;
        ConfirmEmail: Serenity.EmailEditor;
        Password: Serenity.PasswordEditor;
        ConfirmPassword: Serenity.PasswordEditor;
    }

    export class SignUpForm extends Serenity.PrefixedContext {
        static formKey = 'Membership.SignUp';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!SignUpForm.init)  {
                SignUpForm.init = true;

                var s = Serenity;
                var w0 = s.StringEditor;
                var w1 = s.EmailEditor;
                var w2 = s.PasswordEditor;

                Q.initFormType(SignUpForm, [
                    'DisplayName', w0,
                    'Email', w1,
                    'ConfirmEmail', w1,
                    'Password', w2,
                    'ConfirmPassword', w2
                ]);
            }
        }
    }
}
